contracts/interchain-token-service/src/contract/execute.rs

```diff
@@ -21,12 +21,12 @@ pub enum Error {
     FailedItsContractRegistration(ChainNameRaw),
     #[error("failed to deregister its contract for chain {0}")]
     FailedItsContractDeregistration(ChainNameRaw),
-    #[error("failed to execute message")]
-    FailedExecuteMessage,
     #[error("failed to query nexus")]
     NexusQueryError,
-    #[error("storage error")]
-    StorageError,
+    #[error("state error")]
+    StateError,
```

Comment:
```suggestion
    #[error(transparent)]
    State(#[from] state::Error)
```
Does this work?

Reply:
not sure that works with `change_context`, but either way, it should be `State` and not `StateError`

Reply:
yeah `transparent` doesn't work very cleanly with `change_context`

Reply:
Instead of `.change_context(Error::State)?` shall we do `.map_err(Error::State)?` if we want to wrap it. Not a strong opinion either way